tests/aequilibrae/conftest.py

```diff
@@ -12,20 +12,64 @@
 from ..data import siouxfalls_project
 
 
+@pytest.fixture
+def create_project(create_project_session):
+    projects = []
+
+    def _create_project(name=None, source_dir=siouxfalls_project):
+        project = create_project_session(name=name, source_dir=source_dir, teardown=False)
+        projects.append(project)
+        return project
+
+    yield _create_project
+    for project in projects:
+        project.close()
```

Comment:
something feels a little off here - why are we maintaining two lists of created projects (one here and one in create_project_session). I'm guessing one is persistent across the entire test session and one is only within a given test... but it seems over complicated the way it is setup. Maybe I just don't understand the utility though.

Reply:
That is mainly for performance reasons. Ideally we would want a fresh project for every test. However, creating a new project takes some time, so when tests only do read actions on a project we can reuse a project through several tests. Looking at `test_field_editor.py` for example, we went from a few large tests that create a project per test, to many smaller tests. If each of those tests would create a fresh project, the test suite would slow down significantly. Instead we create one project for the full test suite, and clean it up (reset it) for those tests that actually modify it.

It is setup in this way because teardown of a fixture only happens once the scope of the fixture has ended, and we want to close the projects as soon as they're no longer needed. We could do all project's closing at session teardown, but then we'd have resources building up (dangling db connection etc) throughout the test session.

Having said that, I can try and refactor these fixtures to make them a little bit less confusing

Reply:
I've refactored these fixtures. Is it better now? or too many nested functions? 😅

Reply:
Looks much better - I think i was unclear in my comment - I understood why you would want different scopes, but the previous impl called from one to the other and had that janky flag to say whether things added should be cleaned up.... this is much better.